flake.nix

```diff
@@ -0,0 +1,59 @@
+{
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixos-23.11";
+
+    fenix = {
+      url = "github:nix-community/fenix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+
+    flake-compat = {
```

Comment:
Is this required? Can we assume flake support by now? Related: Is this workaround needed in every project?

Reply:
Nix flakes really improves reproducibility, but it is still (since years…) an experimental features. Therefore it is not available by default in NixOS nor Nixpkgs. The compat is here for non-flakes people to still be able to spawn Nix shells:

- With Flakes support: `nix develop` spawns a shell based on `flake.{nix,lock}`
- Without Flakes support (native Nix): `nix-shell` spawns a shell based on `shell.nix` (or, if missing, `default.nix`).